src/VirtualShield.cpp

```diff
@@ -36,14 +36,24 @@
 #if defined(ARDUINO) && ARDUINO > 100
   // Define the serial port that is used to talk to the virtual shield.
   #define VIRTUAL_SERIAL_PORT0 Serial
-
-  // If it has dual serial ports (Leonardo), prefer the second one (pins 0, 1) for bluetooth.
-  #if defined(__AVR_ATmega32U4__)
+  
+  // Force a link to printf float functionality. There is a bug where some boards do not link this properly.
+  asm(".global _printf_float");
+  
+  // For 32 bit boards, strlen_PF takes a char*. Otherwise, it needs a uint_farprt_t.
+  #if defined(ARDUINO_SAMD_MKR1000) || defined(_VARIANT_ARDUINO_101_X_)
```

Comment:
Instead of defining a MACRO, you could `#define uint_farptr_t`, so it evaluates to nothing, and retains its original type of `const char *`